_studio/mfx_lib/decode/mpeg2/src/mfx_mpeg2_decode.cpp

```diff
@@ -1,3293 +1,1145 @@
-
-    Reset(0);
-}
-
-VideoDECODEMPEG2InternalBase::~VideoDECODEMPEG2InternalBase()
-{
-}
-
-mfxStatus VideoDECODEMPEG2InternalBase::Reset(mfxVideoParam *par)
-{
-    if (m_FrameAllocator)
-    {
-        for (mfxU32 i = 0; i < DPB; i++)
-        {
-            if (mid[i] >= 0)
-            {
-                m_FrameAllocator->DecreaseReference(mid[i]);
-                mid[i] = -1;
-            }
-        }
-
-        m_FrameAllocator->Reset();
-    }
-
-    prev_index      = -1;
-    curr_index      = -1;
-    next_index      = -1;
-    display_index   = -1;
-    display_order   =  0;
-    dec_frame_count = 0;
-    dec_field_count = 0;
-    last_frame_count = 0;
-    display_frame_count = 0;
-    cashed_frame_count = 0;
-    skipped_frame_count = 0;
-    last_timestamp   = 0;
-    last_good_timestamp = 0.0;
-
-    m_found_SH = false;
-    m_found_RA_Frame = false;
-    m_first_SH = true;
-    m_new_bs   = true;
-
-    ResetFcState(m_fcState);
-
-    m_task_num = 0;
-    m_prev_task_num = -1;
-
-    for(int i = 0; i < NUM_FRAMES; i++)
-    {
-        m_frame[i].DataLength = 0;
-        m_frame[i].DataOffset = 0;
-        m_frame_in_use[i] = false;
-        m_time[i] = (mfxF64)(-1);
-    }
-
-    m_frame_curr = -1;
-    m_frame_free = -1;
-    m_frame_constructed = true;
-
-    memset(m_last_bytes,0,NUM_REST_BYTES);
-
-    m_resizing = false;
-    m_InitPicStruct = 0;
-
-    m_vdPar.info.framerate = 0;
-
-    if (par)
-    {
-        m_vPar = *par;
-
-        m_vdPar.info.stream_type = UMC::MPEG2_VIDEO;
-        m_vdPar.info.clip_info.width = par->mfx.FrameInfo.Width;
-        m_vdPar.info.clip_info.height = par->mfx.FrameInfo.Height;
-
-        m_vdPar.info.aspect_ratio_width = par->mfx.FrameInfo.AspectRatioW;
-        m_vdPar.info.aspect_ratio_height = par->mfx.FrameInfo.AspectRatioH;
-
-        mfxU32 frameRateD = par->mfx.FrameInfo.FrameRateExtD;
-        mfxU32 frameRateN = par->mfx.FrameInfo.FrameRateExtN;
-
-        if (0 != frameRateD && 0 != frameRateN)
-        {
-            m_vdPar.info.framerate = (double) frameRateN / frameRateD;
-            m_isFrameRateFromInit = true;
-        }
-
-        m_vdPar.info.color_format = GetUmcColorFormat((mfxU8)par->mfx.FrameInfo.ChromaFormat);
-
-        if (UMC::NONE == m_vdPar.info.color_format)
-        {
-            return MFX_ERR_INVALID_VIDEO_PARAM;
-        }
-
-        m_vdPar.lFlags |= par->mfx.DecodedOrder ? 0 : UMC::FLAG_VDEC_REORDER;
-        m_vdPar.lFlags |= UMC::FLAG_VDEC_EXTERNAL_SURFACE_USE;
-
-        switch(par->mfx.TimeStampCalc)
-        {
-            case MFX_TIMESTAMPCALC_TELECINE:
-                m_vdPar.lFlags |= UMC::FLAG_VDEC_TELECINE_PTS;
-                break;
-
-            case MFX_TIMESTAMPCALC_UNKNOWN:
-                break;
-
-            default:
-                return MFX_ERR_INVALID_VIDEO_PARAM;
-        }
-
-        m_vdPar.numThreads = (mfxU16)(par->AsyncDepth ? par->AsyncDepth : m_pCore->GetAutoAsyncDepth());
-
-        if (m_InitH || 
```

Comment:
"is should" sounds incorrect, maybe bShouldUpdate?

Reply:
Ok